zetaclient/chains/bitcoin/signer/outbound_data.go

```diff
@@ -0,0 +1,137 @@
+package signer
+
+import (
+	"fmt"
+	"math"
+	"strconv"
+
+	"github.com/btcsuite/btcd/btcutil"
+	"github.com/pkg/errors"
+	"github.com/rs/zerolog"
+
+	"github.com/zeta-chain/node/pkg/chains"
+	"github.com/zeta-chain/node/pkg/coin"
+	"github.com/zeta-chain/node/pkg/constant"
+	"github.com/zeta-chain/node/x/crosschain/types"
+	"github.com/zeta-chain/node/zetaclient/chains/bitcoin/common"
+	"github.com/zeta-chain/node/zetaclient/compliance"
+)
+
+// OutboundData is a data structure containing necessary data to construct a BTC outbound transaction
+type OutboundData struct {
+	// chainID is the external chain ID
+	chainID int64
+
+	// to is the recipient address
+	to btcutil.Address
+
+	// amount is the amount in BTC
+	amount float64
+
+	// amountSats is the amount in satoshis
+	amountSats int64
+
+	// feeRate is the fee rate in satoshis/vByte
+	feeRate int64
+
+	// txSize is the average size of a BTC outbound transaction
+	// user is charged (in ZRC20 contract) at a static txSize on each withdrawal
+	txSize int64
+
+	// nonce is the nonce of the outbound
+	nonce uint64
+
+	// height is the ZetaChain block height
+	height uint64
+
+	// cancelTx is a flag to indicate if this outbound should be cancelled
+	cancelTx bool
+}
+
+// NewOutboundData creates OutboundData from the given CCTX.
+func NewOutboundData(
+	cctx *types.CrossChainTx,
+	chainID int64,
+	height uint64,
+	minRelayFee float64,
+	logger, loggerCompliance zerolog.Logger,
+) (*OutboundData, error) {
+	if cctx == nil {
+		return nil, errors.New("cctx is nil")
+	}
+	params := cctx.GetCurrentOutboundParam()
+
+	// support gas token only for Bitcoin outbound
+	if cctx.InboundParams.CoinType != coin.CoinType_Gas {
+		return nil, errors.New("can only send gas token to a Bitcoin network")
+	}
+
+	// initial fee rate
+	feeRate, err := strconv.ParseInt(params.GasPrice, 10, 64)
+	if err != nil || feeRate <= 0 {
+		return nil, fmt.Errorf("invalid fee rate %s", params.GasPrice)
+	}
+
+	// use current gas rate if fed by zetacore
+	newRate, err := strconv.ParseInt(params.GasPriorityFee, 10, 64)
+	if err == nil && newRate > 0 && newRate != feeRate {
+		logger.Info().Msgf("use new fee rate %d sat/vB instead of %d sat/vB", newRate, feeRate)
+		feeRate = newRate
+	}
+
+	// check receiver address
+	to, err := chains.DecodeBtcAddress(params.Receiver, params.ReceiverChainId)
+	if err != nil {
+		return nil, errors.Wrapf(err, "cannot decode receiver address %s", params.Receiver)
+	}
+	if !chains.IsBtcAddressSupported(to) {
+		return nil, fmt.Errorf("unsupported receiver address %s", params.Receiver)
```

Comment:
Should show `to` data

Reply:
updated https://github.com/zeta-chain/node/pull/3381/commits/3f5087b2b7a10b24e7ce5831dd65fcdf0f385216